core/services/ardupilot_manager/flight_controller_detector/Detector.py

```diff
@@ -18,89 +19,114 @@ def _is_root() -> bool:
         return os.geteuid() == 0
 
     @staticmethod
-    def detect_navigator_r3() -> Tuple[bool, str]:
-        """Check if navigator R3 is connected using the sensors on the I²C BUS
+    def get_navigator_if_connected() -> Optional[FlightController]:
```

Comment:
This should follow the same function name pattern as `detect_serial_flight_controllers`, like: `detect_navitator_board`